app/components/agenda/agendaitem/agendaitem-decision.js

```diff
@@ -182,6 +184,15 @@ export default class AgendaitemDecisionComponent extends Component {
     }
   });
 
+  loadMayBeEdited = task(async () => {
+    if (this.currentSession.may('manage-decisions') && 
+    !this.isEditingAnnotation && 
+    (this.pieceParts || !this.report) 
+    && this.mayEditDecisionReport) {
+      this.mayBeEdited = true;
+    }
+  });
```

Comment:
I like the idea of having the "may be edited" conditions in a separate function, however, this does not need to be a `async` task because there is nothing to be `await`ed. This makes the code unnecessarily complicated. I think it would be better to see if this code can be reconciled with the already existing `mayEditDecisionReport` getter and use that one instead. Also, be careful, the condition is not 3 times the same, there is a difference in `isEditingAnnotation` (vs `isEditingTreatment`).